components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,380 @@
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+)
+
+const errStrAccuracy = "_accuracy_err"
+
+var model = resource.DefaultModelFamily.WithModel("merged")
+
+// Config is the config of the merged movement_sensor model.
+type Config struct {
+	Position           []string `json:"position,omitempty"`
+	Orientation        []string `json:"orientation,omitempty"`
+	CompassHeading     []string `json:"compass_heading,omitempty"`
+	LinearVelocity     []string `json:"linear_velocity,omitempty"`
+	AngularVelocity    []string `json:"angular_velocity,omitempty"`
+	LinearAcceleration []string `json:"angular_acceleration,omitempty"`
+}
+
+// Validate validates the merged model's configuration.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	deps = append(deps, cfg.Position...)
+	deps = append(deps, cfg.Orientation...)
+	deps = append(deps, cfg.CompassHeading...)
+	deps = append(deps, cfg.LinearVelocity...)
+	deps = append(deps, cfg.AngularVelocity...)
+	deps = append(deps, cfg.LinearAcceleration...)
+	return deps, nil
+}
+
+type merged struct {
+	resource.Named
+	logger golog.Logger
+
+	mu sync.Mutex
+
+	ori     movementsensor.MovementSensor
+	pos     movementsensor.MovementSensor
+	compass movementsensor.MovementSensor
+	linVel  movementsensor.MovementSensor
+	angVel  movementsensor.MovementSensor
+	linAcc  movementsensor.MovementSensor
+}
+
+func init() {
+	resource.Register(
+		movementsensor.API, model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{
+			Constructor: newMergedModel,
+		})
+}
+
+func newMergedModel(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (
+	movementsensor.MovementSensor, error,
+) {
+	m := merged{
+		logger: logger,
+	}
+
+	if err := m.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+func (m *merged) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	firstGoodSensorWithProperties := func(
+		deps resource.Dependencies, names []string, logger golog.Logger,
+		mprops *movementsensor.Properties, propname string,
+	) (movementsensor.MovementSensor, error) {
+		// check if the config names and dependencies have been passed at all
+		if len(names) == 0 || deps == nil {
+			return nil, nil
+		}
+
+		for _, name := range names {
+			ms, err := movementsensor.FromDependencies(deps, name)
+			if err != nil {
+				logger.Debugf("error getting sensor %v from dependencies", ms.Name().ShortName())
+				continue
+			}
+
+			props, err := ms.Properties(ctx, nil)
+			if err != nil {
+				logger.Debugf("error in getting sensor %v properties", ms.Name().ShortName())
+				continue
+			}
+
+			// found check that the sensor matches the properties that  want
+			// if it doesn't, skip it and go on to the next sensor in the list
```

Comment:
This comment needs some cleanup. There are two spaces between "that want" and I suspect a word like "we" goes in there. Maybe remove "found" from the beginning? I think there's a period after "want" and "if" should be capitalized, but I could be wrong about that.